File: pinmaker/solutions/martinvl.java
/*
@EXPECTED_RESULTS@: CORRECT
*/

import java.util.Scanner;

public class martinvl {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);

        int d = in.nextInt();
        int m = in.nextInt();
        int n = in.nextInt();

        System.out.printf("%08d\n", keanucci(d, m, n));
    }

    static int keanucci(int d, int m, int n) {
        if (n == 0) {
            return d;
        }

        if (n == 1) {
            return m;
        }

        int pprev = d;
        int prev = m;

        for (int i = 2; i <= n; ++i) {
            int cur = (pprev + prev) % (int)1e8;
            pprev = prev;
            prev = cur;
        }

        return prev;
    }
}

File: pinmaker/solutions/martinvl.py
# @EXPECTED_RESULTS@: CORRECT

from sys import stdin

def keanucci(d, m, n):
    if n == 0:
        return d

    if n == 1:
        return m

    pprev = d
    prev = m

    for i in xrange(n-1):
        cur = (pprev + prev) % int(1e8)
        pprev = prev
        prev = cur

    return prev

d, m, n = map(int, stdin.read().split())
print '%08d' % (keanucci(d, m, n) % int(1e8))
